### ISSUE TITLE
Accessing relation in model causes dirty state to change to transient

### ISSUE THREAD

**Issue by nsossonko** [phalcon/cphalcon#11042]
Code:

```php
  class Parent extends \Phalcon\Mvc\Model {
    public $id;
    public function initialize() {
      $this->hasMany('id', 'Child', 'parent', array(
        'foreignKey' => array('action' => Relation::ACTION_RESTRICT),
        'alias'  => 'relatedchilds'));
    }
  }
  class Child extends \Phalcon\Mvc\Model {
    public $parent;
    public $id;
    public function initialize() {
      $this->belongsTo('parent', 'Parent', 'id', array(
          'alias' => 'relatedparent',
          'foreignKey' => array('message' => 'Must have parent')
      ));
    }
  }

  $children = Child::find(); // get all child records
  foreach ($children as $child) {
    echo "Initial state: {$child ->getDirtyState()} <br />";
    $parent = $child ->relatedparent;
    echo "After accessing parent - state: {$child ->getDirtyState()} <br />";
  }
```

Is this by design? If so, why? This is causing us big headaches in our application, so any help would be appreciated.

**Comment by andresgutierrez** [comment]
I'm not sure how can this work:

```php
$child = Child::find(); // this returns a resultset
  echo "Initial state: {$child ->getDirtyState()} <br />";
  $parent = $child ->relatedparent; // you can't do this
```

**Comment by nsossonko** [comment]
What's the problem, I'm grabbing the related parent of the child...?

**Comment by andresgutierrez** [comment]
Child::find() returns a resultset you can access the related parent on a resultset.

**Comment by nsossonko** [comment]
Oh, I see what you're saying...it's a mistake in the code. Just assume there's a foreach loop there and this is done on one of the actual models...sorry it was pseudo-code

**Comment by nsossonko** [comment]
Should be clearer now, I updated the code

**Comment by nsossonko** [comment]
Btw, is there any reason the dirty state should change if the property is being set to an object? That itself also doesn't make sense to me...this record is in the database and is going to be updated, why should it be in TRANSIENT state? Setting any other property on the record doesn't change the dirty state...

**Comment by nsossonko** [comment]
This is fixed by PR #11055
